### DISCUSSION TITLE
HyperHDR 20.0.0.0beta2

### DISCUSSION THREAD

**Discussion by awawa-dev** [awawa-dev/HyperHDR#778] [General]
Full list of changes below, with details in the attached links. Update is also available in HyperHDR Debian/Ubuntu APT repository: <a href="https://awawa-dev.github.io/">link</a>.

HyperHDR: v20beta2: 
- SD images are available again and are based on Debian Bullseye, so they will not fit RPi 5: use the official Raspberry Pi installer for this board. It is explained here: <a href="https://github.com/awawa-dev/HyperHDR/issues/699">link1</a> 
- Fix: Fedora uses xz as the package name, not xz-utils <a href="https://github.com/awawa-dev/HyperHDR/pull/771">#771</a> Thanks @hsmalley
- Fix LUT loading bug to reduce memory usage by 96MB <a href="https://github.com/awawa-dev/HyperHDR/pull/766">#766</a>
- Fix background music effect <a href="https://github.com/awawa-dev/HyperHDR/pull/761">#761</a>
- Fix typo <a href="https://github.com/awawa-dev/HyperHDR/pull/762">#762</a> Thanks @laurens94
- Yeelight: Wizard now supports more models, workaround for music-mode reset, auto-resume support <a href="https://github.com/awawa-dev/HyperHDR/pull/750">#750</a> Thanks @ratawhisk
- Fix verbose command line option <a href="https://github.com/awawa-dev/HyperHDR/pull/723">#723</a>
- Add native build for Apple M1 / M2 (arm64) architecture <a href="https://github.com/awawa-dev/HyperHDR/pull/755">#755</a>
- New features: disable LEDs/components on startup <a href="https://github.com/awawa-dev/HyperHDR/pull/737">#737</a>
- New features: stop processing when user has locked the system <a href="https://github.com/awawa-dev/HyperHDR/pull/737">#737</a>
- Force HyperSerial detection <a href="https://github.com/awawa-dev/HyperHDR/pull/732">#732</a> Thanks @alex-013
- Fix Linux DBus sleep/wakeup handler
- Fix pipewire grabber for non-contiguous DMA memory & EGL <a href="https://github.com/awawa-dev/HyperHDR/pull/711">#711</a> Thanks @Mic92
- Update build scripts and include Ubuntu 24.04 LTS
- Fixes for v20beta1 <a href="https://github.com/awawa-dev/HyperHDR/pull/705">#705</a>

HyperHDR: v20beta1:  
- Refactoring. Migration to C++ Smart Pointers. Move almost all main objects management under control of modern C++ smart pointers, fully control complicated life-cycle and ownership of HyperHDR components <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Windows/DirectX: use hardware acceleration (pixel & vertex shaders) for software capturing. <a href="https://github.com/awawa-dev/HyperHDR/issues/688">Fixes issue #688</a> → <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Windows/DirectX: fix overexposed HDR stream with pixel shader. <a href="https://github.com/awawa-dev/HyperHDR/issues/689">Fixes issue #689</a> → <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Windows/DirectX: autodetect HDR mode.<a href="https://github.com/awawa-dev/HyperHDR/issues/690">Fixes issue #690</a> → <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Automatic database backup before applying structure changes when installing new version, json backups are saved in standard HyperHDR configuration home directory <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Smoothing clock is integrated directly within led driver: no more two separated & unsynchronized clocks for smoothing and led driver, also much more precise refresh times. <a href="https://github.com/awawa-dev/HyperHDR/discussions/512">Discussed here</a> → <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- HyperHDR driver for WLED has been switched to WLED Audio-Reactive-Led-Strip (WARLS) protocol removing old 490 LEDs limit https://github.com/Aircoookie/WLED/wiki/UDP-Realtime-Control → <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- switch libCEC to mainstream since version 6.x is widely available now <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Linux: audio stream auto-resume on EPIPE error <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Improved headers including for faster compilation <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- JSON API security vulnerability fixed <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Add video buffer memory caching to Pipewire grabber <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Fixed the LUT calibration page by removing potential scrollbars that could interfere with the final result <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Support for Pico (rp2040) boards using HyperSPI <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Fix over 500LEDs limit for HyperSPI when RPi5 is used <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Improved installers script <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Address CodeQL findings <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Fix logging tab that can be empty when opened and add button to copy logs <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- ArtNet LED Controller: add toggle to prevent pixel color data split across two artnet universes. <a href="https://github.com/awawa-dev/HyperHDR/issues/657">Fixes issue #657</a> → <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Ability to change MQTT Topic. <a href="https://github.com/awawa-dev/HyperHDR/pull/619">Fixes issue #619</a> → <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- MQTT: auto-resume. <a href="https://github.com/awawa-dev/HyperHDR/issues/570">Fixes issue #570</a> → <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- MQTT ability to execute multiply commands in one request. <a href="https://github.com/awawa-dev/HyperHDR/issues/642">Fixes issue #642</a> → <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Each MQTT subcommand will return full response in the overall result at HyperHDR/JsonAPI/response <a href="https://github.com/awawa-dev/HyperHDR/pull/691">#691</a>
- Fix typo in LED strip name <a href="https://github.com/awawa-dev/HyperHDR/pull/670">#670</a> Thanks @fluchfux
- Pipewire DMA & EGL hardware support (Wayland/x11 grabber) <a href="https://github.com/awawa-dev/HyperHDR/pull/556">#556</a>
- Upgrade Fedora 38 to 39, Ubuntu 23.04 to 23.10 <a href="https://github.com/awawa-dev/HyperHDR/pull/667">#667</a>
- Colorized logs <a href="https://github.com/awawa-dev/HyperHDR/pull/640">#640</a>
- Fix macOS build <a href="https://github.com/awawa-dev/HyperHDR/pull/638">#638</a> #671 #672
- Update language file's. Thanks @AstaRom <a href="https://github.com/awawa-dev/HyperHDR/pull/617">#617</a>
- Make bonjour use logging utilities. Thanks @nurikk <a href="https://github.com/awawa-dev/HyperHDR/pull/529">#529</a>
- New interface: removed ancient Font Awesome 4 (so 2017...) Migrate to SVG: Bootstrap Icons and Google Material Icons/Symbols <a href="https://github.com/awawa-dev/HyperHDR/pull/605">#605</a>
- New device discovery service <a href="https://github.com/awawa-dev/HyperHDR/pull/605">#605</a>
- Stability improvements & bug fixing <a href="https://github.com/awawa-dev/HyperHDR/pull/605">#605</a>
- Fix QT5.15 issue when building on Windows <a href="https://github.com/awawa-dev/HyperHDR/pull/605">#605</a>
- Add new effect: Rainbow Waves <a href="https://github.com/awawa-dev/HyperHDR/pull/607">#607</a>
- MQTT: hostname address resolver. Thanks @lougovsk <a href="https://github.com/awawa-dev/HyperHDR/pull/586">#586</a>
- V4L: allow setting of V4L device input, fixes #537. Thanks @Thinner77 <a href="https://github.com/awawa-dev/HyperHDR/pull/540">#540</a>
- Update flatbuffers to v23.5.26 <a href="https://github.com/awawa-dev/HyperHDR/pull/596">#596</a>
- Pause the USB grabber when all LEDs are off to reduce load <a href="https://github.com/awawa-dev/HyperHDR/pull/608">#608</a>
- Add support for HD108 led strip, 16/48bits colors <a href="https://github.com/awawa-dev/HyperHDR/pull/527">#527</a>
- Use optional system libs for compiling <a href="https://github.com/awawa-dev/HyperHDR/pull/605">#541</a>
- Update mbedtls to 3.4.0 <a href="https://github.com/awawa-dev/HyperHDR/pull/589">#589</a>
- Add Ubuntu support to HyperHDR and Github Pages APT repository <a href="https://github.com/awawa-dev/HyperHDR/pull/522">#522</a>
- New JsonAPI method to calculate average color of selected instance <a href="https://github.com/awawa-dev/HyperHDR/pull/611">#611</a>
- Workaround for critical Rpi udev bug affecting serial ports <a href="https://github.com/awawa-dev/HyperHDR/pull/583">#583</a>
- Add Arch Linux support <a href="https://github.com/awawa-dev/HyperHDR/pull/520">#520</a>
- Fix chrome/edge fullscreen detection <a href="https://github.com/awawa-dev/HyperHDR/pull/519">#519</a>
- Add automatic LUT downloader & installer <a href="https://github.com/awawa-dev/HyperHDR/pull/568">#568</a>
- Allow LUT calibration when no USB grabber is present <a href="https://github.com/awawa-dev/HyperHDR/pull/518">#518</a>
- Live update WLED/udpraw LEDs number <a href="https://github.com/awawa-dev/HyperHDR/pull/564">#564</a>
- Add support for Raspberry Pi Pico (Rp2040) HyperSerialPico <a href="https://github.com/awawa-dev/HyperHDR/pull/560">#560</a>
- Do not use TurboJPEG for RGB during calibration <a href="https://github.com/awawa-dev/HyperHDR/pull/508">#508</a>
- Update Github actions/checkout@v3 <a href="https://github.com/awawa-dev/HyperHDR/pull/403">#403</a>
- glibc requirements for DEB installer <a href="https://github.com/awawa-dev/HyperHDR/pull/402">#402</a>
- Remove folders nesting <a href="https://github.com/awawa-dev/HyperHDR/pull/363">#363</a>
- New JSON API method to update smoothing parameters <a href="https://github.com/awawa-dev/HyperHDR/pull/352">#352</a>
- Update Github actions/download@v3 <a href="https://github.com/awawa-dev/HyperHDR/pull/372">#372</a>
- Remove depraciated ::set-output from Github Action <a href="https://github.com/awawa-dev/HyperHDR/pull/390">#390</a>
- Upgrade Bootstrap to 5.2.3 / improved 'video live preview' on smartphones <a href="https://github.com/awawa-dev/HyperHDR/pull/336">#336</a>
- Update language file's. Thanks @AstaRom <a href="https://github.com/awawa-dev/HyperHDR/pull/605">#505</a>

<hr /><em>This discussion was created from the release <a href='https://github.com/awawa-dev/HyperHDR/releases/tag/v20.0.0.0beta2'>HyperHDR 20.0.0.0beta2</a>.</em>

**Comment by Duese123** [comment]
Is there no arm7 built for raspberry?

**Reply by awawa-dev** [reply]
No, use aarch64 (arm64) release if you are using 64-bit OS which offer best performance or arm6 for 32-bit legacy OS. You can compile HyperHDR yourself for 32-bit armv7h, but it will still be slower than the 64-bit version. #694
